packages/web/src/components/pool/exchange-rate-graph-content/ExchangeRateGraphContent.tsx

```diff
@@ -0,0 +1,177 @@
+import ChartScopeSelectTab from "@components/common/chart-scope-select-tab/ChartScopeSelectTab";
+import LineGraph from "@components/common/line-graph/LineGraph";
+import { CHART_DAY_SCOPE_TYPE } from "@constants/option.constant";
+import { useTheme } from "@emotion/react";
+import useComponentSize from "@hooks/common/use-component-size";
+import { useWindowSize } from "@hooks/common/use-window-size";
+import { TokenModel } from "@models/token/token-model";
+import { TokenExchangeRateGraphResponse } from "@repositories/token/response/token-exchange-rate-response";
+import { DEVICE_TYPE } from "@styles/media";
+import { getLocalizeTime, parseDate } from "@utils/chart";
+import { useMemo, useState } from "react";
+import PairRatio from "../../common/pair-ratio/PairRatio";
+import { ExchangeRateGraphContentHeader, ExchangeRateGraphContentWrapper, ExchangeRateGraphXAxisWrapper } from "./ExchangeRateGraphContent.styles";
+
+interface ExchangeRateGraphContentProps {
+  tokenA: TokenModel;
+  tokenB: TokenModel;
+  feeTier: string;
+  onSwap?: (swap: boolean) => void
+  data?: TokenExchangeRateGraphResponse
+}
+
+const calculateMiddleIndices = (totalLabels = 0, countXAxis = 0) => {
+  const indices = new Set<number>();
+  // Helper function to add indices
+  const addIndices = (start: number, end: number) => {
+    const mid = Math.floor((start + end) / 2);
+    if (!indices.has(mid)) {
+      indices.add(mid);
+      if (indices.size < countXAxis) {
+        // Add midpoint of the left subarray
+        addIndices(start, mid - 1);
+        // Add midpoint of the right subarray
+        addIndices(mid + 1, end);
+      }
+    }
+  };
+
+  // Always include the first and last labels
+  indices.add(0);
+  indices.add(totalLabels - 1);
+
+  // Begin by adding the middle of the entire array
+  addIndices(0, totalLabels - 1);
+
+  // Convert to array and sort to ensure the correct order
+  return Array.from(indices).sort((a, b) => b - a);
+};
```

Comment:
Can you tell me the purpose of this function?
If you want to get equally spaced values, it seems like a good idea to check out the function.

The result of the function behavior.
```
INPUT : calculateMiddleIndices(20,6)
OUTPUT: [19,14,9,6,4,2,1,0]
```

Would the expression below produce a different result? 
```
const spacing = (totalLabels - 1) / (countXAxis - 1);
Array.from({length: countXAxis}, (_, index) => Math.floor(spacing * index))
```

Reply:
I inherited it from old code so please let me check again

Reply:
For this case i re-used the earlier code, i tried the new approach and it worked, thanks.